src/train.jl

```diff
@@ -1,404 +1,6 @@
-end
-
-struct BBO
-  method::Symbol
-end
-
-BBO() = BBO(:adaptive_de_rand_1_bin)
-
-function sciml_train(loss, _θ, opt::BBO = BBO(), data = DEFAULT_DATA;lower_bounds, upper_bounds,
-                      cb = (x...)-> false, maxiters = get_maxiters(data), kwargs...)
-  local x, cur, state
-  cur,state = iterate(data)
-
-  function _cb(trace)
-    cb_call = cb(decompose_trace(trace),x...)
-    if !(typeof(cb_call) <: Bool)
-      error("The callback should return a boolean `halt` for whether to stop the optimization process. Please see the sciml_train documentation for information.")
-    end
-    if cb_call == true
-      BlackBoxOptim.shutdown_optimizer!(trace) #doesn't work
-    end
-    cur,state = iterate(data,state)
-    cb_call
-  end
-
-  _loss = function (θ)
-    x = loss(θ,cur...)
-    first(x)
-  end
-
-  bboptre = BlackBoxOptim.bboptimize(_loss;Method = opt.method, SearchRange = [(lower_bounds[i], upper_bounds[i]) for i in 1:length(lower_bounds)], MaxSteps = maxiters, CallbackFunction = _cb, CallbackInterval = 0.0, kwargs...)
-
-  Optim.MultivariateOptimizationResults(opt.method,
-                                        [NaN],# initial_x,
-                                        best_candidate(bboptre), #pick_best_x(f_incr_pick, state),
-                                        best_fitness(bboptre), # pick_best_f(f_incr_pick, state, d),
-                                        bboptre.iterations, #iteration,
-                                        bboptre.iterations >= maxiters, #iteration == options.iterations,
-                                        true, # x_converged,
-                                        0.0,#T(options.x_tol),
-                                        0.0,#T(options.x_tol),
-                                        NaN,# x_abschange(state),
-                                        NaN,# x_abschange(state),
-                                        true,# f_converged,
-                                        0.0,#T(options.f_tol),
-                                        0.0,#T(options.f_tol),
-                                        NaN,#f_abschange(d, state),
-                                        NaN,#f_abschange(d, state),
-                                        true,#g_converged,
-                                        0.0,#T(options.g_tol),
-                                        NaN,#g_residual(d),
-                                        false, #f_increased,
-                                        nothing,
-                                        maxiters,
-                                        maxiters,
-                                        0,
-                                        true,
-                                        NaN,
-                                        bboptre.elapsed_time,
-                                        NamedTuple())
+function sciml_train(loss, θ, opt, adtype::DiffEqBase.AbstractADType = GalacticOptim.AutoZygote(); kwargs...)

```

Comment:
this doesn't handle the data iterator?

Reply:
Replacing the 4th argument with the data iterator will likely cause many cases to fail, like the MNIST one. Instead, pass the data iterator to GalacticOptim (it has a data iterator... right?) and just implicitly hardcode this to AutoZygote (which is essentially what sciml_train had done before).

Reply:
GalacticOptim doesn't have a data iterator, I'll take a look at the mnist example and make the fixes